src/main/java/com/codessquad/qna/web/QuestionController.java

```diff
@@ -45,4 +52,48 @@ public String showOneQuestion(@PathVariable long qnaId, Model model) {
         model.addAttribute("Qna", qnaRepository.findById(qnaId).orElseThrow(NoQuestionException::new));
         return "qna/show";
     }
+
+    @GetMapping("{id}/form")
+    public String editQna(@PathVariable long id, Model model, HttpSession session) {
+        Qna qna = qnaRepository.findById(id).orElseThrow(NoUserException::new);
+        User sessionUser = HttpSessionUtils.getUserFromSession(session);
+
+        if (!HttpSessionUtils.isLoginUser(session)) {
+            return "redirect:/users/login";
+        }
+        assert sessionUser != null;
+        if (!qna.checkWriter(sessionUser.getUserId())) {
+            throw new IllegalStateException("자신의 질문만 수정할 수 있습니다.");
+        }
+
+        model.addAttribute("qna", qna);
+
+        return "qna/updateForm";
+    }
+
+    @PostMapping("{id}/form")
+    public String update(@PathVariable long id, Qna updateQna, HttpSession session) {
+        Qna qna = qnaRepository.findById(id).orElseThrow(NoUserException::new);
+        qna.update(updateQna);
+        qnaRepository.save(qna);
+        return "redirect:/";
+    }
+
+    @DeleteMapping("{id}/delete")
```

Comment:
```suggestion
    @DeleteMapping("{id}")
```

RESTful한 Url 설계에 대해서 한 번 학습해봅시다.

Reply:
이전에도 알려주셨는데 또 같은 실수를 했네요..🥲 주의하겠습니다!